File: app/src/main/java/com/ja_cabili/passifi/Event.java
package com.ja_cabili.passifi;

public class Event {
    public int id;
    public String title;
    public int avatar;
    public String description;
    public String date;
    public String time;
    public String location;
    public String organizer;
    public String organizer_email;
    public String organizer_approval;
    public String user_id;
    public String invite_code;
    public int total_attendees;
    public int current_attendees;

    public Event(String title, int avatar, String description, String date, String time, String location, String organizer, String organizer_email, String organizer_approval, String user_id, String invite_code, int total_attendees, int current_attendees, int id) {
        this.title = title;
        this.description = description;
        this.avatar = avatar;
        this.date = date;
        this.time = time;
        this.location = location;
        this.organizer = organizer;
        this.organizer_email = organizer_email;
        this.organizer_approval = organizer_approval;
        this.user_id = user_id;
        this.invite_code = invite_code;
        this.total_attendees = total_attendees;
        this.current_attendees = current_attendees;
        this.id = id;
    }
}

File: app/src/main/java/com/ja_cabili/passifi/model/Event.kt
package com.ja_cabili.passifi.model

import android.util.Log
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Locale

data class Event(
    val id: Int,
    val avatar: String?,
    val description: String,
    val title: String,
    val date: String,
    val time: String,
    val location: String,
    val organizer: String,
    val organizer_email: String,
    val organizer_approval: String,
    val user_id: Int,
    val invite_code: String,
    val current_attendees: Int?,
    val total_attendees: Int?,
    val guests: List<Guest>?,
) {
    fun getFormattedDate(): String {
        return try {
            val inputFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
            val outputFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
            val parsedDate = inputFormat.parse(date)
            Log.d("Event", "Formatted date: ${outputFormat.format(parsedDate!!)}")
            outputFormat.format(parsedDate)
        } catch (e: ParseException) {
            Log.d("Event", "Error parsing date: $e")
            date // Return original date in case of parsing error
        }
    }
}
